src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const length = this.length;
+      let element = this[i];
+
+      if (i === length - 1) {
```

Comment:
```suggestion
      let element = this[i];

      if (i === this.length - 1) {
```
Actually, there's no need in this variable. You use it only once and you could just use `this.length` instead